gateway/api/views/jobs.py

```diff
@@ -116,29 +159,26 @@ def result(self, request, pk=None):  # pylint: disable=invalid-name,unused-argum
         tracer = trace.get_tracer("gateway.tracer")
         ctx = TraceContextTextMapPropagator().extract(carrier=request.headers)
         with tracer.start_as_current_span("gateway.job.result", context=ctx):
-            saved = False
-            attempts_left = 10
-            while not saved:
-                if attempts_left <= 0:
-                    return Response(
-                        {"error": "All attempts to save results failed."}, status=500
-                    )
+            author = self.request.user
+            job = self.jobs_repository.get_job_by_id(pk)
+            if job is None:
+                logger.info("Job [%s] nor found", pk)
+                return Response(
+                    {"message": f"Job [{pk}] nor found"},
+                    status=status.HTTP_404_NOT_FOUND,
+                )
 
-                attempts_left -= 1
-
-                try:
-                    job = self.get_object()
-                    job.result = json.dumps(request.data.get("result"))
-                    job.save()
-                    saved = True
-                except RecordModifiedError:
-                    logger.warning(
-                        "Job [%s] record has not been updated due to lock. Retrying. Attempts left %s",  # pylint: disable=line-too-long
-                        job.id,
-                        attempts_left,
-                    )
-                    continue
-                time.sleep(1)
+            can_access = JobAccessPolocies.can_save_result(author, job)
```

Comment:
I would change probably the name of the variable to something like: `can_save_result`. Thinking in can be confusing with the `can_access` method, WDYT @paaragon ?